Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4784,6 +4785,46 @@ void WebPageProxy::disconnectFramesFromPage()
         mainFrame->webProcessWillShutDown();
 }
 
+void WebPageProxy::disownOpener()
+{
+    if (!m_process->processPool().configuration().processSwapsOnWindowOpenWithOpener())
+        return;
+    if (m_initialWindow && m_opener) {
+        auto process = WebProcessProxy::processForIdentifier(m_initialWindow->processIdentifier);
+        process->send(Messages::WebProcess::RemoveRemoteWindow(*m_opener), 0);
+    }
+    m_opener = std::nullopt;
+}
+
+void WebPageProxy::didCreateWindow(uint64_t frameID, GlobalWindowIdentifier&& windowIdentifier, std::optional<GlobalWindowIdentifier> openerID)
+{
+    if (!m_process->processPool().configuration().processSwapsOnWindowOpenWithOpener())
+        return;
+    if (!m_mainFrame || m_mainFrame->frameID() != frameID)
+        return;
+    if (!m_initialWindow)
+        m_initialWindow = windowIdentifier;
```

Comment:
Assuming initial window means "main frame window" here, I think we should rename to "m_mainFrameWindow" for clarity.